### ISSUE TITLE
Node 10 has changed how fs.stat works

### ISSUE THREAD

**Issue by imurchie** [tschaub/mock-fs#236]
After `mock-fs` does its magic calling code ends up with `Stats` objects like
```js
Stats {
  dev: undefined,
  mode: undefined,
  nlink: undefined,
  uid: undefined,
  gid: undefined,
  rdev: undefined,
  blksize: undefined,
  ino: undefined,
  size: undefined,
  blocks: undefined,
  atimeMs: undefined,
  mtimeMs: undefined,
  ctimeMs: undefined,
  birthtimeMs: undefined,
  atime: Invalid Date,
  mtime: Invalid Date,
  ctime: Invalid Date,
  birthtime: Invalid Date }
```

To reproduce: `npm run test`.

**Comment by gucong3000** [comment]
+1

**Comment by szll** [comment]
+1

**Comment by BugDiver** [comment]
When are we getting a fix for this?

**Comment by tschaub** [comment]
Thanks for the report @imurchie.  I've put together a fix in #237.

**Comment by imurchie** [comment]
Amazing! Thank you @tschaub!
